core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -56,13 +56,26 @@
 
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
 
-    {% if unique_key is not none %}
-    delete from {{ target }}
-    where ({{ unique_key }}) in (
-        select ({{ unique_key }})
-        from {{ source }}
-    );
-    {% endif %}
+    {% if unique_key %}
+        {% if unique_key is sequence and unique_key is not mapping and unique_key is not string %}
```

Comment:
If it's a sequence, can it also be a mapping or string?  Why is it not enough to just check if it is a sequence?

Reply:
I believe we defined it this way to prevent attempts of use of dictionaries which sequences I think tie into as well and to skip the iteration part of this conditional if it did just turn out to be a single string. is my understanding.